cmd/user-agent/src/pages/GetVC.vue

```diff
@@ -5,119 +5,272 @@ SPDX-License-Identifier: Apache-2.0
+                        }
+                    ).catch(err => {
+                        data = err
+                        console.log('failed to create presentation, errMsg:', err)
+                    })
+                }
+
+                if (data.retry) {
+                    this.errors.push(data.retry)
+                    return
+                }
+
+                // Call Credential Handler callback
+                this.credentialEvent.respondWith(new Promise(function (resolve) {
+                    return resolve({
+                        dataType: "Response",
+                        data: data
+                    });
+                }))
+            },
+            getSelectedCredentials: async function () {
+                if (this.selectedVCs.length == 0) {
+                    return {retry: "Please select at least one credential"}
+                }
+
+                try {
+                    let vcs = []
+                    for (let selectedVC of this.selectedVCs) {
+                        const resp = await this.aries.verifiable.getCredential({
+                            id: this.savedVCs[selectedVC].key
+                        })
+                        vcs.push(JSON.parse(resp.verifiableCredential))
+                    }
+                    return {vcs: vcs}
+                } catch (e) {
+                    return e
+                }
+            },
+            loadIssuers: async function () {
+                await this.aries.vdri.getDIDRecords().then(
+                    resp => {
+                        const data = resp.result
+                        if (!data || data.length == 0) {
+                            this.errors.push("No issuers found to select, please create an issuer")
+                            return
+                        }
+
+                        this.issuers = []
+                        this.selectedIssuer = 0
+
+                        data.forEach((item, id) => {
+                            this.issuers.push({id: id, name: item.name, key: item.id})
+                        })
+                    })
+                    .catch(err => {
+                        this.errors.push(err)
+                    })
+            },
+            cancel: async function () {
+                // Call Credential Handler callback
+                this.credentialEvent.respondWith(new Promise(function (resolve) {
+                    return resolve({
+                        dataType: "Response",
+                        data: "DID Auth cancelled"
+                    });
+                }))
+            },
+            authorize: async function () {
                 let data
-                await this.aries.verifiable.getCredential({
-                    id: this.selectedVC
+                await this.aries.verifiable.generatePresentation({
+                    presentation: {
+                        "@context": "https://www.w3.org/2018/credentials/v1",
```

Comment:
Why are we doing this ? is it related to creating the presentation automatically when user doesn't select any vc ??

Reply:
This is for DID Auth only. Response should be  an empty VP signed with given domain & challenge.